sdk/textanalytics/Azure.AI.TextAnalytics/src/Transforms.Legacy.cs

```diff
@@ -0,0 +1,644 @@
+
+        internal static DetectLanguageResultCollection ConvertToDetectLanguageResultCollection(Legacy.LanguageResult results, IDictionary<string, int> idToIndexMap)
+        {
+            var detectedLanguages = new List<DetectLanguageResult>(results.Errors.Count);
+
+            //Read errors
+            foreach (var error in results.Errors)
+            {
+                detectedLanguages.Add(new DetectLanguageResult(error.Id, ConvertToError(error.Error)));
+            }
+
+            //Read languages
+            foreach (var language in results.Documents)
+            {
+                detectedLanguages.Add(new DetectLanguageResult(language.Id, ConvertToDocumentStatistics(language.Statistics), ConvertToDetectedLanguage(language)));
+            }
+
+            detectedLanguages = SortHeterogeneousCollection(detectedLanguages, idToIndexMap);
+
+            return new DetectLanguageResultCollection(detectedLanguages, ConvertToBatchStatistics(results.Statistics), results.ModelVersion);
+        }
+
+        #endregion
+
+        #region AnalyzeSentiment
+
+        internal static TextSentiment ConvertToTextSentiment(Legacy.Models.SentenceSentimentValue sentiment) =>
+            sentiment switch
+            {
+                Legacy.Models.SentenceSentimentValue.Neutral => TextSentiment.Neutral,
+                Legacy.Models.SentenceSentimentValue.Positive => TextSentiment.Positive,
+                Legacy.Models.SentenceSentimentValue.Negative => TextSentiment.Negative,
+                _ => throw new NotSupportedException($"The sentence sentiment, { sentiment }, is not supported for conversion.")
+            };
+
+        internal static TextSentiment ConvertToTextSentiment(Legacy.Models.TokenSentimentValue sentiment) =>
+            sentiment switch
+            {
+                Legacy.Models.TokenSentimentValue.Mixed => TextSentiment.Neutral,
+                Legacy.Models.TokenSentimentValue.Positive => TextSentiment.Positive,
+                Legacy.Models.TokenSentimentValue.Negative => TextSentiment.Negative,
+                _ => throw new NotSupportedException($"The token sentiment, { sentiment }, is not supported for conversion.")
+            };
+
+        internal static TextSentiment ConvertToTextSentiment(Legacy.Models.DocumentSentimentValue sentiment) =>
+            sentiment switch
+            {
+                Legacy.Models.DocumentSentimentValue.Neutral => TextSentiment.Neutral,
+                Legacy.Models.DocumentSentimentValue.Positive => TextSentiment.Positive,
+                Legacy.Models.DocumentSentimentValue.Negative => TextSentiment.Negative,
+                _ => throw new NotSupportedException($"The document sentiment, { sentiment }, is not supported for conversion.")
+            };
+
+        //TODO:  Check this conversion.  Seems to match: https://github.com/Azure/azure-sdk-for-net/blob/Azure.AI.TextAnalytics_5.2.0-beta.2/sdk/textanalytics/Azure.AI.TextAnalytics/src/DocumentSentiment.cs#L56
```

Comment:
@maririos:   I could use your expertise here.  From what I can tell, this is the same conceptual transform that was being performed.   But, the existing transformation is odd and I'm not sure that I'm interpreting it correctly.

Reply:
Could you remind me what the need to add this code here instead of leaving it as is?

Reply:
The types changed and the structure didn't line up in the way that I expected.    Did I overlook something that would make that unnecessary?

Reply:
what I see missing is how are we treating the [TargetRelation](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/textanalytics.json#L1386) .
This type has a json pointer that correlates the assessment to a specific sentence which is what the code in [here](https://github.com/Azure/azure-sdk-for-net/blob/Azure.AI.TextAnalytics_5.2.0-beta.2/sdk/textanalytics/Azure.AI.TextAnalytics/src/SentenceSentiment.cs#L86) is doing.

Reply:
@maririos:  Thank you very much for all of your help in sorting out the issues here.  Based on our conversation, I refactored and I think we've now got the correct transform.